generate/src/contract.rs

```diff
@@ -118,11 +118,18 @@ pub(crate) fn expand_contract(args: &Args) -> Result<TokenStream> {
                 Self{ instance }
             }
 
-            /// Retrieve the undelying instance being used by this contract.
-            pub fn instance(&self) -> &#ethcontract::DynInstance {
+            /// Retrieve a reference the undelying base `ethcontract::Instance`
+            /// being used by this contract.
+            pub fn base_instance(&self) -> &#ethcontract::DynInstance {
```

Comment:
Not a big fan of exposing the entire `base_instance`. Are there other use cases for it? I'd slightly prefer exposing the defaults on the main contract instance as it keeps the abstraction exposed to the developer simple and intact.

Reply:
Right now it is also needed for the fallback function since we don't generate the method for it. I do agree that a contract should have its own default configuration API though.

Reply:
How is the fallback function exposed at the moment? Wouldn't we be able to expose this on the generated instance as well?

Reply:
It's not really exposed at all at the moment actually. You would need to use the instance to create a `MethodBuilder` and then get the inner `TransactionBuilder` to set the data. There is an open issue to make this nicer. We should be able to remove the `base_instance()` then.

Reply:
There are other ways to get around this now. I will remove the `base_instance(_mut)` accessor and add ones specifically for the defaults. And instance can be created like this worst case scenario:
```
Instance::at(web3, contract.address(), &Contract::artifact().abi)
```